plugin/src/main/java/org/opensearch/ml/indices/MLIndicesHandler.java

```diff
@@ -108,8 +78,67 @@ public void initMLIndexIfAbsent(String indexName, String mapping, ActionListener
             }
         } else {
             log.info("index:{} is already created", indexName);
-            listener.onResponse(true);
+            if (indexMappingUpdated.containsKey(indexName) && !indexMappingUpdated.get(indexName).get()) {
+                shouldUpdateIndex(indexName, index.getVersion(), ActionListener.wrap(r -> {
+                    if (r) {
+                        // return true if should update index
+                        client
+                            .admin()
+                            .indices()
+                            .putMapping(
+                                new PutMappingRequest().indices(indexName).source(mapping, XContentType.JSON),
+                                ActionListener.wrap(response -> {
+                                    if (response.isAcknowledged()) {
+                                        indexMappingUpdated.get(indexName).set(true);
+                                        listener.onResponse(true);
+                                    } else {
+                                        listener.onFailure(new MLException("Failed to update index: " + indexName));
+                                    }
+                                }, exception -> {
```

Comment:
Should we set indexMappingUpdated to false explicitly when getting an exception? I am not sure if there is a chance that we get an exception after setting indexMappingUpdated to true. So I recommend set to false here.

Reply:
The `indexMappingUpdated` flag is to make sure we only update the index mapping for first request.  `indexMappingUpdated` was initialized as false, check line 44-45. We will set the flag as true if

1. Index doesn't exist. When create index, it will use latest mapping.
2. Index exists but its mapping is already latest (for example, some other request/node already updated the index mapping), then no need to update it. 
3. Index mapping is old, and we updated its mapping successfully.

For case 1 and 2, they map to line 104-105.
For case3, it maps to line 92.

When we set `indexMappingUpdated` as true. That means the index mapping already updated. No need to set `indexMappingUpdated` as false and update again.

Reply:
Thanks for the explanation.